src/ucp/rma/flush.c

```diff
@@ -449,6 +466,22 @@ static void ucp_worker_flush_ep_flushed_cb(ucp_request_t *req)
     ucp_request_put(req);
 }
 
+static UCS_F_ALWAYS_INLINE void
+ucp_worker_flush_req_set_next_ep(ucp_request_t *req, ucp_ep_ext_gen_t *ep_ext)
+{
+    ucp_worker_h worker = req->flush_worker.worker;
+    ucp_ep_h ep         = ucp_ep_from_ext_gen(ep_ext);
+
+    req->flush_worker.next_ep = ep_ext;
+
+    if (&ep_ext->ep_list != &worker->all_eps) {
+        /* Increment UCP EP reference counter to avoid destroying UCP EP while
+         * it is being scheduled to be flushed */
+        ucp_ep_add_ref(ep);
+        UCP_EP_ASSERT_COUNTER_INC(&ep->num_flush_inprog);
```

Comment:
move it inside ucp_ep_flush_internal

Reply:
why we should move it there? this increment counter for next EP, but `ucp_ep_flush_internal()` works with the current one

Reply:
looking at ep->num_flush_inprog counter name, i would expect it to mean "how many flush operations are in progress for this endpoint"
otherwise it's very confusing, maybe better just remove this counter

Reply:
we can't remove this counter, we rely on this counter when checking EP reference counter.
EP reference counter should be `num_flush + num_discard + 1` or `num_flush + num_discard`

Reply:
gave a more clear name to the counter